### PR TITLE
ft: update crr metrics e2e tests

### PR DESCRIPTION
**Pull request by philipyoo**

### CODE REVIEW

File: tests/backbeat/Using.md
Comment (bennettbuchanan) [inline tests/backbeat/Using.md:107]: 👍

File: tests/backbeat/Using.md
Comment (bennettbuchanan) [inline tests/backbeat/Using.md]: Where are we using an AWS source bucket in the API tests that we need to define this?

File: tests/backbeat/Using.md
Comment (bennettbuchanan) [inline tests/backbeat/Using.md:104]: I'd also add the steps needed to create the Azure destination location. I think these steps would be the same as those for creating Azure resources for CRR tests—probably just copying over those steps will do. 😸

File: tests/backbeat/tests/api/metrics.js
Comment (bennettbuchanan) [inline tests/backbeat/tests/api/metrics.js]: Seems like maybe we would want to iterate through `metricTypes` and comparing them to the object body so that we can catch a case where a metric is missing from the response.

File: tests/backbeat/tests/api/metrics.js
Comment (bennettbuchanan) [inline tests/backbeat/tests/api/metrics.js]: Looks like it could be cleaner to use two describe statements because these `beforeEach` and `afterEach` hooks are not needed for the tests that are checking the API route properties (e.g. the test `should return all metrics for path ${path}`). That is, those tests should pass without having created a bucket and setup replication on it.

File: tests/backbeat/tests/api/metrics.js
Comment (bennettbuchanan) [inline tests/backbeat/tests/api/metrics.js]: Let's use the `doWhilst` here to wait for the expected response. I'm comfortable saying the test failed if the metrics aren't there by the time the `REPLICATION_TIMEOUT` is exceeded.

File: tests/backbeat/tests/api/metrics.js
Comment (bennettbuchanan) [inline tests/backbeat/tests/api/metrics.js]: Maybe you can use a unique key name for this test (assuming it maintains the common key prefix) such that you can guarantee the expected throughput size.

File: tests/backbeat/tests/api/metrics.js
Comment (bennettbuchanan) [inline tests/backbeat/tests/api/metrics.js]: Looks like this test case is already handled in https://github.com/scality/Zenko/blob/development/0.9/tests/backbeat/tests/api/objectMonitor.js#L39.

File: tests/backbeat/tests/api/metrics.js
Comment (bennettbuchanan) [inline tests/backbeat/tests/api/metrics.js]: This PR https://github.com/scality/Zenko/pull/221 should handle this test case as well. Probably a refactor is in order to move all the backbeat API tests into a single file.

**Review by bennettbuchanan — COMMENTED** [review]
Nice! ❤️these new test cases.

File: tests/backbeat/Using.md
Comment (philipyoo) [reply tests/backbeat/Using.md:107]: ya i noticed we use both ENDPOINT and HOST in setup :O

File: tests/backbeat/tests/api/metrics.js
Comment (philipyoo) [reply tests/backbeat/tests/api/metrics.js]: for now, i'll remove the object-level tests from this file

File: tests/backbeat/tests/api/metrics.js
Comment (bennettbuchanan) [reply tests/backbeat/tests/api/metrics.js]: As per our conversation, the object-level metrics don't apply here, so the above comment is `void` 😂.

File: tests/backbeat/tests/api/metrics.js
Comment (bennettbuchanan) [inline tests/backbeat/tests/api/metrics.js]: I noticed a race condition here in another test run: since Azure does not have versioning, if `scalityUtils.deleteVersionedBucket` manages to replicate the deletion of the object version before `scalityUtils.deleteAllBlobs` tries to clean up the destination container, then a `BlobNotFound` error is thrown and `afterEach` fails. We can avoid this case by doing the cleanup in series, where we delete the blobs before deleting the source bucket.

File: tests/backbeat/tests/api/metrics.js
Comment (bennettbuchanan) [inline tests/backbeat/tests/api/metrics.js]: Since this is a one-to-many setup we can just use the util `waitUntilReplicated`—i.e. no need to compare the destination object.

File: tests/backbeat/tests/api/metrics.js
Comment (bennettbuchanan) [inline tests/backbeat/tests/api/metrics.js:189]: Minor: looks like this will generally be `'0.00'`, maybe we could consider putting N objects to the point that the value is increased in the API response.

File: tests/backbeat/tests/api/metrics.js
Comment (JianqinWang) [inline tests/backbeat/tests/api/metrics.js:108]: missing ';'

File: tests/backbeat/tests/api/metrics.js
Comment (bennettbuchanan) [inline tests/backbeat/tests/api/metrics.js:133]: We still want to skip this one? LGTM 😸

**Review by bennettbuchanan — APPROVED** [review]

File: tests/backbeat/tests/api/metrics.js
Comment (philipyoo) [reply tests/backbeat/tests/api/metrics.js:133]: i was experiencing some issues still where metrics data from other tests were still included. For now going to commit it with the plans to continue testing

Comment (bennettbuchanan) [reply tests/backbeat/tests/api/metrics.js:133]: Okay, still good as a manual test for now. 👍

File: tests/backbeat/tests/api/metrics.js
Comment (philipyoo) [reply tests/backbeat/tests/api/metrics.js:189]: do you mean replicate N number of objects and each time check that throughput increases?

Comment (bennettbuchanan) [reply tests/backbeat/tests/api/metrics.js:189]: Was thinking that replicating >= 5 objects would result in the API responding with a `count` floating point > `0.00`. It's possibly overkill since having the `size` property increase probably suffices.

Comment (philipyoo) [reply tests/backbeat/tests/api/metrics.js:189]: Ah I see, I'll look into adding it with the fix when re-enabling this test case 👍

**Pull request merged**
